libraries/rush-lib/src/logic/operations/OperationExecutionRecord.ts

```diff
@@ -185,6 +195,17 @@ export class OperationExecutionRecord implements IOperationRunnerContext, IOpera
     return TERMINAL_STATUSES.has(this.status);
   }
 
+  public get executedOnThisAgent(): boolean {
+    if (!this._context.cobuildConfiguration) {
+      return this.cobuildRunnerId === undefined;
+    }
+    return (
+      // this can happen if this property is retrieved before `beforeResult` is called.
+      this.cobuildRunnerId !== undefined &&
+      this._context.cobuildConfiguration?.cobuildRunnerId === this.cobuildRunnerId
+    );
+  }
```

Comment:
I'd rather just have a mutable property for `agentId` or whatever, that way `OperationExecutionRecord` doesn't have any dependency on `CobuildConfiguration`.